### PR TITLE
Add BCV61/62 and generic dual transistor version

### PR DESCRIPTION
**Pull request by evanshultz**
Image:
image

Notes:
- I'm undecided on text placement. Probably to the right is better since wires will naturally go vertical?
- Is having the base connections this way acceptable? It's weirdly off-grid but looks like the datasheet.
- I did not add and entry into device.dcm since they're generic. I can if desired.
- The FPFilter may be too specific for the generic part. Please advise.

Datasheets:
http://www.nxp.com/documents/data_sheet/BCV61.pdf
http://www.nxp.com/documents/data_sheet/BCV62.pdf

### CODE REVIEW

**Comment by jkriege2** [comment]
Hi!

a few things to change please:
1. add description to the new generic symbols in `device.lib` (i.e. put them also into `device.dcm`)
2. I'm not sure whether the FPFilter is needed for the generic symbol ... I would rather say no, as you don't really know in which package they might also come
3. Please correct line-endings to UNIX (LF-only).

Thanks,
JAN

**Comment by evanshultz** [comment]
1. Done. Oops.
2. Yeah, I see that. But a 4 pin connector wouldn't be suitable. Perhaps we wait a couple days for feedback in Issue #986?
3. I've been seeing a lot of talk about line endings recently, and I use Windows. But what I don't see is what to do about it, just that it was a problem or that it was done. How to correct line endings?

**Comment by evanshultz** [comment]
3. Got it. Notepad++ can convert automatically. Just a sec.

**Comment by evanshultz** [comment]
Just above the comment box there is a conflicts section but I'm not seeing that it pertains to my PR. In fact, it's not clear to me what the conflict is. Is it line endings? When I copy part of the committed text to Notepad++ and view line endings, they're all only LF which is correct. Is it something else?

**Comment by poeschlr** [comment]
> Just above the comment box there is a conflicts section but I'm not seeing that it pertains to my PR.
At the bottom of this conversation stuff i see this:
screenshot from 2017-02-25 18-43-01


> it's not clear to me what the conflict is.
someone else has changed the files since your branch deviated from the master branch.
(The last commit to the transistors.lib file was the changing of some footprint filters by @jkriege2)
To resolve it: merge in the changes done on this files.

**Comment by jkriege2** [comment]
The conflicts mean that `transistors.lib` and `transistors.dcm` have changed in the master. This prevents a merge. So please pull/merge the master into you branch. That should solve the conflicts!

**Comment by evanshultz** [comment]
I think that should do it, but the PR isn't being checked. How to trigger that? I will close and reopen.

**Comment by evanshultz** [comment]
Hmm.... I pulled the current master for each `transistors` file from the repo, then re-added my two new parts, and still errors throughout the file. Any tips on what I'm doing wrong?

File: library/transistors.dcm [hunk 1]
```diff
@@ -82,7 +82,7 @@ $ENDCMP
 #
 $CMP 2SC1945
 D 6A Ic, 80V Vce, Silicon 27MHz RF Power Transistors NPN, TO-220
-K RF Power Transistor NPN
```
Comment (poeschlr) [inline library/transistors.dcm]: Example for a changed line ending or removed whitespace

File: library/transistors.lib [hunk 2]
```diff
@@ -634,8 +657,8 @@ ENDDEF
 #
 DEF BC807 Q 0 0 Y N 1 F N
 F0 "Q" 200 75 50 H V L CNN
-F1 "BC807" 200 0 50 H V L CNN
-F2 "TO_SOT_Packages_SMD:SOT-23" 200 -75 50 H I L CIN
```
Comment (poeschlr) [inline library/transistors.lib]: Here is another example of a changed line ending

**Comment by poeschlr** [comment]
The curse of the line endings it seems.
And there are also some whitespace characters removed.
(See my comments above)

**Comment by poeschlr** [comment]
> Hmm.... I pulled the current master for each transistors file from the repo, then re-added my two new parts, and still errors throughout the file. Any tips on what I'm doing wrong?

To sync with master do the following.
git checkout master
git add remote upstream https://github.com/KiCad/kicad-library.git
git fetch upstream
git pull upstream/master
git checkout [your branch]
git merge master

check if it worked with git status
possibly you need to git add -u
git commit -m [a commit message]

**Comment by jkriege2** [comment]
To make this simpler:
remove `transistors.*` from this PR and open a fresh new one from a new branch branched off from an up-to-date master and add the special transistors there ... sometimes it's easier to start all over than putting in a lot of work to fix a crooked branch ...

Best,
JAN

PS: If you do, just post the second PR here, so I can review them together to speed things up a bit ;-)!

**Pull request closed without merging**